BackgroundJobs/BackgroundExchangeService.cs

```diff
@@ -24,18 +24,21 @@ public BackgroundExchangeService(ILogger<BackgroundExchangeService> logger, ISer
             _backgroundModel = backgroundModel;
         }
 
-        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+        protected override async Task ExecuteAsync(CancellationToken cancellationToken)
         {
-            while (_backgroundModel.IsActive)
+            if (_backgroundModel.IsActive)
```

Comment:
the better approach is to exit in case that service is not active. Benefits: less code, better readability

if (!_backgroundModel.IsActive){
return;
}